src/node.h

```diff
@@ -55,6 +55,7 @@ struct cmark_node {
   struct cmark_node *last_child;
 
   void *user_data;
+  CMarkNodeUserDataFreeFunc user_data_free_func;
```

Comment:
Would it be possible to only add this field to node types that actually need it?